src/app-logic/url-handling.js

```diff
@@ -886,7 +889,7 @@ const _upgraders = {
   },
   [4]: (
     processedLocation: ProcessedLocationBeforeUpgrade,
-    profile: Profile
+    profile?: Profile
```

Comment:
I wonder if we could add a type for the `_upgraders` variable that would catch this error?

Reply:
Good idea, I'm adding a commit which does that.